sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/util/Utility.java

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.search.documents.implementation.util;
+
+import com.azure.core.serializer.json.jackson.JacksonJsonSerializerBuilder;
+import com.azure.core.util.serializer.JacksonAdapter;
+import com.azure.core.util.serializer.JsonSerializer;
+import com.azure.core.util.serializer.SerializerAdapter;
+import com.azure.search.documents.implementation.serializer.SerializationUtil;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public final class Utility {
+    /**
+     * Helper class to initialize the SerializerAdapter.
+     * @return The SerializeAdapter instance.
+     */
+    public static SerializerAdapter initializeSerializerAdapter() {
+        JacksonAdapter adapter = (JacksonAdapter) JacksonAdapter.createDefaultSerializerAdapter();
+
+        ObjectMapper mapper = adapter.serializer();
+        SerializationUtil.configureMapper(mapper);
+
+        return adapter;
+    }
+
+    public static JsonSerializer creatDefaultJsonSerializerInstance() {
+        JacksonAdapter adapter = (JacksonAdapter) initializeSerializerAdapter();
+        return new JacksonJsonSerializerBuilder().serializer(adapter.serializer()).build();
```

Comment:
We're in a tricky spot for Search here as we've used a non-standard `JacksonAdapter` to handle serialization before introducing a pluggable serializer. I'm wondering for Search whether the custom `JacksonAdapter` should be used to handle serialization when `JsonSerializer` is `null`. That way we retain previous functionality when a customer upgrades but then we don't need to have this which will force a customer to include `azure-core-serializer-json-jackson` on their classpath, potentially never allowing `azure-core-serializer-json-gson` to work implicitly (not certain this would be the case but I don't exactly know how the classpath ordering is determined for a service provider).

If this change is made in places where serialization of customer objects take place it would need to be updated to the following:

```java
if (jsonSerializer == null) {
  // Use Search JacksonAdapter
} else {
  // Use customer JsonSerializer
}
```

cc: @srnagar @JonathanGiles

Reply:
The method here is dealing with default case, which means customer does not pass us anything in builder. The logic you put here is inside of builder.
We can simply take a parameter of customer serializer and change as what you proposed.

Reply:
The method is to use when customer does not  pass us JsonSerializer.
We have to use JsonSerializer API  for workflow. If return JacksonAdapter, this is how we hook up with JsonSerializer. Only JacksonJsonSerializerBuilder provide a way to take ObjectMapper.

Reply:
Talked offline. We will use search specific JacksonAdapter as a default serializer if customer do not pass any serializer in.